starlite/_signature/parsing.py

```diff
@@ -125,14 +128,18 @@ def get_type_annotation_from_plugin(
 
 
 def parse_fn_signature(
-    fn: AnyCallable, plugins: list[SerializationPluginProtocol], dependency_name_set: set[str]
+    fn: AnyCallable,
+    plugins: list[SerializationPluginProtocol],
+    dependency_name_set: set[str],
+    owner: Controller | Router | None = None,
```

Comment:
Why pass owner here rather than simply pass in the annotations? It's better to inject these values from outside than to make this function more complex.

Reply:
I use the owner directly, for both identifying that it is a subclass of `GenericController` and for calling a method on it.

I can remove the method call, but I still need to know that we are dealing with a signature from a handler that is defined on a generic controller before I can start attempting to swap out type vars in the signature.

Reply:
The method called on the owner is `owner.get_parameter_annotation()` which is the one that detects if a type var exists and swaps it out for the type it has been narrowed with. So to remove everything I'm using the owner for here then I need to pass in a boolean indicating whether the owner is a generic controller, and the type that the generic controller has been narrowed with which is otherwise stored on the generic controller subtype.